packages/build/src/plugins_core/blobs_upload/index.ts

```diff
@@ -0,0 +1,62 @@
+import { getDeployStore } from '@netlify/blobs'
+import pMap from 'p-map'
+
+import { addErrorInfo } from '../../error/info.js'
+import { log, logError } from '../../log/logger.js'
+
+import { getKeysToUpload, getBlobsDir, getFileWithMetadata, anyBlobsToUpload } from './utils.js'
+
+const coreStep = async function ({
+  debug,
+  logs,
+  deployId,
+  buildDir,
+  constants: { PUBLISH_DIR, SITE_ID, NETLIFY_API_TOKEN, API_URL },
+}) {
+  const blobStore = getDeployStore({ siteID: SITE_ID, deployID: deployId, token: NETLIFY_API_TOKEN, apiURL: API_URL })
+  const blobsDir = getBlobsDir({ buildDir, publishDir: PUBLISH_DIR })
+  const keys = await getKeysToUpload(blobsDir)
+
+  // We checked earlier, but let's be extra safe
+  if (keys.length === 0) {
+    log(logs, 'No blobs to upload to deploy store.')
+    return {}
+  }
+
+  log(logs, `Uploading ${keys.length} blobs to deploy store...`)
+
+  const uploadBlob = async (key) => {
+    if (debug) {
+      log(logs, `- Uploading blob ${key}`, { indent: true })
+    }
+    const { data, metadata } = await getFileWithMetadata(blobsDir, key)
+    await blobStore.set(key, data, { metadata })
+  }
+
+  try {
+    await pMap(keys, uploadBlob, { concurrency: 10 })
+  } catch (err) {
+    logError(logs, `Error uploading blobs to deploy store: ${err.message}`)
+
+    const error = new Error(`Failed while uploading blobs to deploy store`)
+    addErrorInfo(error, { type: 'blobsUploadError' })
```

Comment:
I wasn't sure about this and the`type` here. It's not defined in https://github.com/netlify/build/blob/4511447230ae5b582821b40499ae29d97af0aeae/packages/build/src/error/type.js#L42

Should something change? /cc @JGAntunes I think you're the expert in this area.

Reply:
Update after a chat: I'll remove the `addErrorInfo` and rely on the default `coreStep` added by the system.